pkg/executor/internal/mpp/local_mpp_coordinator.go

```diff
@@ -343,6 +351,130 @@ func (c *localMppCoordinator) fixTaskForCTEStorageAndReader(exec *tipb.Executor,
 	return nil
 }
 
+// taskZoneInfoHelper used to help reset exchange executor's same zone flags
+type taskZoneInfoHelper struct {
+	allTiflashZoneInfo map[string]string
+	// exchangeZoneInfo is used to cache one mpp task's zone info:
+	// key is executor id, value is zone info array
+	// for ExchangeSender, it's target tiflash nodes' zone info; for ExchangeReceiver, it's source tiflash nodes' zone info
+	exchangeZoneInfo map[string][]string
+	tidbZone         string
+	currentTaskZone  string
+	isRoot           bool
+}
+
+func (h *taskZoneInfoHelper) init(store kv.Storage) {
+	tikvStore, ok := store.(helper.Storage)
+	if !ok {
+		return
+	}
+	h.tidbZone = config.GetGlobalConfig().Labels[placement.DCLabelKey]
+	cache := tikvStore.GetRegionCache()
+	allTiFlashStores := cache.GetTiFlashStores(tikv.LabelFilterNoTiFlashWriteNode)
+	h.allTiflashZoneInfo = make(map[string]string, len(allTiFlashStores))
+	for _, tiflashStore := range allTiFlashStores {
+		tiflashStoreAddr := tiflashStore.GetAddr()
+		if tiflashZone, isSet := tiflashStore.GetLabelValue(placement.DCLabelKey); isSet {
+			h.allTiflashZoneInfo[tiflashStoreAddr] = tiflashZone
```

Comment:
`allTiflashZoneInfo` seems to be repeated initialized in multi zoneHelpers. Maybe we can init `allTiflashZoneInfo` in one place for only once.

Reply:
Good point, I'll change it.